src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -318,6 +324,79 @@ def fuzz(self,
     return LibFuzzerCommon.fuzz(self, corpus_directories, fuzz_timeout,
                                 artifact_prefix, additional_args)
 
+class FuchsiaQemuLibFuzzerRunner(new_process.ProcessRunner,LibFuzzerCommon):
+  """libFuzzer runner (when Fuchsia is the target platform)."""
+  def __init__(self, executable_path, default_args=None):
+    # This code assumes the following layout for files in the storage bucket:
+    # * multiboot.bin
+    # * fvm.blk
+    # * fuchsia-ssh.zbi
+    # * qemu/*
+    # * .ssh/*
+    local_path = os.getcwd() + "/"
+    resources_path = local_path + "fuchsia_on_clusterfuzz_resources_v1"
+
+    try:
+      os.mkdir(resources_path)
+    except OSError as e:
+      if e.errno == errno.EEXIST:
+        pass
+    subbed_fuchsia_gsutil_command = [param.replace("{fuchsia_resources_path}", environment.get_value('FUCHSIA_RESOURCES_PATH', ''))
+                                     .replace("{local_resources_path}", resources_path) for param in constants.FUCHSIA_GSUTIL_COMMAND]
+    subprocess.call(subbed_fuchsia_gsutil_command)
+    qemu_path = resources_path + "/qemu-for-fuchsia/bin/qemu-system-x86_64"
+    os.chmod(qemu_path, 0o550)
+    kernel_path = resources_path + "/multiboot.bin"
+    os.chmod(kernel_path, 0o644)
+    pkey_path = resources_path + "/.ssh/pkey"
+    os.chmod(pkey_path, 0o400)
+    sharefiles_path = resources_path + "/qemu-for-fuchsia/share/qemu"
+    initrd_path = resources_path + "/fuchsia-ssh.zbi"
+    os.chmod(initrd_path, 0o644)
+    drive_path = resources_path + "/fuchsia.qcow2"
+    os.chmod(drive_path, 0o644)
+
+    # TODO: Add a mechanism for choosing portnum dynamically.
+    portnum = "56339"
+
+    self.subbed_qemu_base_command = [param.replace("{qemu}", qemu_path)
```

Comment:
I have a suggested improvement for this pattern you use.
Make `FUCHSIA_QEMU_COMMAND_TEMPLATE` one big string and then call `FUCHSIA_QEMU_COMMAND_TEMPLATE.format(kernel=kernel_path, drive=drive_path, ...)`

Reply:
unfortunately, the subprocess API wants to receive commands in the form of a list, rather than one big string.

i could *try* to make it one big string, and call .split(" ") after, but that's harder than it looks-- in the SSH command, "StrictHostKeyChecking no" needs to be one clause (space included), and in the QEMU command "kernel.serial=legacy TERM=dumb" needs to be all one clause (space included).  I could add "dummy" characters for the command to be split along but I worry that's even more confusing.  What do you think?

Reply:
Dummy chars definitely seems like the incorrect approach.
I like the latest approach.